BLE_GattServer_FOTAService/MCUboot/target/application/source/main.cpp

```diff
@@ -0,0 +1,342 @@
+/*
+ * Mbed-OS Microcontroller Library
+ * Copyright (c) 2020-2021 Embedded Planet
+ * Copyright (c) 2020-2021 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+#include "ble/BLE.h"
+#include "ble/gap/Gap.h"
+#include "mbedtls/sha256.h"
+#include "mbedtls/platform.h"
+#include "blockdevice/SlicingBlockDevice.h"
+#include "ble-service-fota/FOTAService.h"
+#include "BlockDeviceFOTAEventHandler.h"
+#include "platform/mbed_power_mgmt.h"
+#include "mbed-trace/mbed_trace.h"
+#include "bootutil/bootutil.h"
+#include "secondary_bd.h"
+
+#define TRACE_GROUP "MAIN"
+
+using namespace std::literals::chrono_literals;
+
+const static char DEVICE_NAME[] = "FOTA";
+
+static events::EventQueue event_queue(/* event count */ 10 * EVENTS_EVENT_SIZE);
+static ChainableGapEventHandler chainable_gap_event_handler;
+static ChainableGattServerEventHandler chainable_gatt_server_event_handler;
+
+/*
+ * Update BlockDevice hook
+ *
+ * This hook returns a pointer to a SlicingBlockDevice of size = 0.5 MiB =
+ * SLOT_SIZE. The SlicingBlockingDevice is backed by the default instance. The
+ * nature of this device depends on the targets default configuration in mbed-
+ * os/targets/targets.json. However, it must be SPIF, QSPIF or OSPIF. For example,
```

Comment:
Why must it be SPIF, QSPIF or OSPIF ?

Reply:
I may have deduced that from somewhere but I can't remember. I'll just remove it.